snippets/javascript/caps-lock-on.md

```diff
@@ -0,0 +1,25 @@
+# Detect Caps Lock Is On
+
+To check if the caps lock is on, you use the `getModifierState()` method of the
+`KeyboardEvent` object:
+
+The `getModifierState()` method returns `true` if a modifier is active;
+otherwise, it returns `false`.
+
+The `event.getModifierState('CapsLock')` can be used to detect if the caps lock
+is on.
+
+## Suppose you have a password field like this
+
+    <input type="password" name="password" id="password" placeholder="Enter a password">
+        <div class="message"></div>
+
+The following shows a warning message if you turn on the caps lock and type the
+password:
+
+    const password = document.querySelector('#password'); const message =
+    document.querySelector('.message');
+
+    password.addEventListener('keyup', function (e) { if
+    (e.getModifierState('CapsLock')) { message.textContent = 'Caps lock is on'; }
+    else { message.textContent = ''; } });
```

Comment:
Tip: you can use indentation to make your code more readable

```js
password.addEventListener('keyup', function (e) { 
    if (e.getModifierState('CapsLock')) {
        message.textContent = 'Caps lock is on'; 
    } 
    else { 
        message.textContent = ''; 
    }
});
```
As an extra tip: You could also take the function out of the addEventListener parameters like this to make it more readable: 

```js
password.addEventListener('keyup', setCapsLockMessage);

function setCapsLockMessage (event){
    if (event.getModifierState('CapsLock')) {
        message.textContent = 'Caps lock is on'; 
    } 
    else { 
        message.textContent = ''; 
    }
}
```

Reply:
@arnochauveau 
I am trying to push the modifications, but I got this error when linting:
MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "````"]

I tried to place (```js ) before the code in the file, but it did not work.

Reply:
Did you add a closing ``` on the end of the code block? You can also try removing the first tab(whitespace before the lines) of the code block. Maybe that interferes.

Reply:
@arnochauveau Finally worked, Thank you